src/io.c

```diff
@@ -2088,6 +2093,14 @@ _dispatch_operation_advise(dispatch_operation_t op, size_t chunk_size)
 	}
 	advise.ra_offset = op->advise_offset;
 	op->advise_offset += advise.ra_count;
+#ifdef __linux__
+	_dispatch_io_syscall_switch(err,
+		readahead(op->fd_entry->fd, advise.ra_offset, advise.ra_count),
+		case EBADF: break; // bad file desciptor
```

Comment:
Ignoring `EBADF` is wrong it should be caught by the `dispatch_assume_zero()` below to log as this is a programming error.